Beta v6.20.1 => v6.20.2

MichaIng opened the pull request:
### Changelog between beta versions v6.20.1 and v6.20.2

#### Changes / Improvements / Optimisations
- DietPi-Config | RPi: Added info to "hifiberry-dacplus" sound device overlay that it supports the AMP2 DAC. Thanks to @Lat Dior for reporting the issue and testing the solution: https://dietpi.com/phpbb/viewtopic.php?t=5396
- DietPi-Software | WireGuard VPN is now available for installation. Is can be easily configured as client or as server, passing either all client traffic through the tunnel, local network access only, or server access only, e.g. as well in combination with Pi-hole, to have remote ad blocking.
- DietPi-RAMlog/Logclear | The internally rotated LetsEncrypt (CertBot) log files are now removed, hourly with DietPi-RAMlog enabled, or when executing DietPi-Logclear manually.

#### Bug Fixes
- DietPi-Software | Deluge: Resolved a visual-only error in systemd unit, since same-line comments are not allowed anymore. Thanks to @Gabba for reporting this issue: https://dietpi.com/phpbb/viewtopic.php?t=5378
- DietPi-Software | No-IP: Fixed an issue, where on "Generic device" (ID 22), the wrong binary could have been installed. As well added failsafe "/usr/local/[bin|etc]" directory pre-creation, in case not present on the system or manually removed by user. Thanks to @walker93 for reporting this issue: https://github.com/Fourdee/DietPi/issues/2426
- DietPi-Config | RPi: gpu_mem_256/512 are now capped when being applied (to prevent value exceeding limits of each board type): https://github.com/Fourdee/DietPi/issues/2417
- DietPi-Config | Resolved a visual-only issue, where WiFi interface starts throw an error message about unsupported power management: https://github.com/Fourdee/DietPi/issues/2198
- DietPi-WiFi-Monitor | Resolved an issue, where DietPi-WiFi-Monitor stuck in a reconnection loop, if no valid WiFi interface was detected or the WiFi adapter is unplugged.
- DietPi-Cloudshell | Resolved an issue where CPU scene failed on VM and other devices that do not support CPU governors: https://github.com/Fourdee/DietPi/issues/2428
- General | Pinebook: Resolved issue with failed apt-get upgrades due to missing armbian-release data file. Also switched to stable kernel packages and kernel updates over APT are now enabled.

MichaIng committed:
v6.20

+ CHANGELOG | Spelling

MichaIng committed:
v6.20

+ CHANGELOG | Add some missing known issues, align with Open Beta issue

MichaIng committed:
v6.20

+ DietPi-Software | Desktops: Create symlinks to $HOME/Desktop after running Create_Desktop_Shared_Items(), because $HOME/Desktop otherwise might not yet exist
+ DietPi-Software | Mate: Add pulse audio sink setting via G_CONFIG_INJECT to avoid double entries

MichaIng committed:
v6.20

+ DietPi-Software | Deluge: Resolve visual-only error on systemd unit, since same-line comments are not allowed anymore

MichaIng committed:
v6.20

+ CHANGELOG | Deluge: Resolved a visual-only error in systemd unit, since same-line comments are not allowed anymore

MichaIng committed:
v6.20

+ DietPi-Config | RPi: Added info to "hifiberry-dacplus" overlay that it supports the AMP2 DAC

MichaIng committed:
v6.20

+ CHANGELOG | RPi: Added info to "hifiberry-dacplus" sound device overlay that it supports the AMP2 DAC

MichaIng committed:
v6.20

+ DietPi-Software | phpBB: Update to v3.2.5
+ DietPi-Software | Minor coding

Commit:
v6.20

+ Always set actual value to gpu_mem_1024, cap the others during set. Scrape from gpu_mem_1024: https://github.com/Fourdee/DietPi/issues/2417

Commit:
v6.20

+ Whoopsie!

Commit:
v6.20

+ Tweaks

MichaIng committed:
v6.20

+ RPi | config.txt: Rearrange video settings: Compatibility first, mode second, resolution third, additional fourth; Add video disable options

MichaIng committed:
v6.20

+ DietPi-PREP | Buster: "systemd" does not depend on "procps" anymore, thus we need to manually install it

MichaIng committed:
v6.20

+ DietPi-Patch | Buster: "systemd" does not depend on "procps" anymore, thus we need to set it manually installed

MichaIng committed:
v6.20

+ DietPi-Set_Hardware | Unsupported_Input_Name/Mode requires to be used with "return" the way it's implemented currently 
+ DietPi-Set_Hardware | RPi i-sabre-k2m: Check, if rebuild is required, by attempting dtoverlay command
+ DietPi-Set_Hardware | G_AGI should be executed before "apt-mark unhold", since it overrides and overwrites hold states anyway (if an actual install/upgrade is done) and unhold fails on non-installed packages
+ DietPi-Set_Hardware | Coding hardening and consistency: Use G_CONFIG_INJECT, where applicable and $(which <cmd>) within systemd units. Check for existence before attempting removal. "systemctl" does not require ".service" ending. Only remove real settings (if desired) instead of every line/comment that matches string. "systemd" does not want to be called "systemD" or with any other capitalization ;).

MichaIng committed:
v6.20 (#2366)

+ Coding | Replace "which" command with builtin "command -v"
+ Coding | Always check "command -v" success directly, instead of indirectly via existing output string
+ DietPi-Autostart | Avoid "bash -c" subshell in systemd unit Exec command, if no output stream redirection is done
+ DietPi-Patch | RPi: G_AGI kernel packages first (which overrides hold state), unhold afterwards, so the latter cannot fail on non-installed package

Commit:
v6.20

+ Minor, correct info at end of apply.

Commit:
v6.20

+ Conflicts

Commit:
v6.20

armbian-release required for kernel package update success.

Commit:
v6.20

- General | Pinebook: Resolved issue with failed apt-get upgrades due to missing armbian-release data file. Also switched to stable kernel packages and kernel updates over APT are now enabled.

MichaIng committed:
v6.20

+ DietPi-Software | No-IP: Check CPU architectures instead of device models, which can lead to wrong result on generic device
+ DietPi-Software | No-IP: Failsafe: Pre-create "/usr/local/bin" and "/usr/local/etc", required to place/create binary and config file
+ DietPi-Software | No-IP: Add config file dependency to systemd unit, to avoid it failing until "noip2 -C" initial setup has finished
+ DietPi-Software | No-IP: Remove the config file as well on uninstall

MichaIng committed:
v6.20

+ DietPi-Config | No-IP: Pre-create config directory, failsafe
+ DietPi-Config | No-IP: Use "No-IP" spelling to match the official one

MichaIng committed:
v6.20

+ CHANGELOG | No-IP: Fixed an issue, where on "Generic device" (ID 22), the wrong binary could have been installed

MichaIng committed:
v6.20

+ DietPi-Sync | Fix wrong tmp log file path variable

MichaIng committed:
v6.20 (#2424)

+ Coding: Replace "awk" by "mawk"
+ Coding: Merge "grep" and "mawk" where applicable
+ Coding: (m)awk /pattern/ does not allow variables. Use -v pat="$var" '$0~pat' to achieve this instead.
+ Coding: Use direct exit code check where applicable, instead of indirect check via "$?"
+ Coding: When assigning a single variable or command substitution to another variable, no double quotes are required to preserve line breaks and spaces. Only when adding as command or command argument, this is required.
+ Coding: Use DietPi-Globals exit trap addition G_EXIT_CUSTOM() to remove temp file and unset arrays
+ Coding: Further minor optimisations

MichaIng committed:
v6.20 (#2420)

+ DietPi-Software | WireGuard: Unmasking kernel packages is not required, since overridden by apt-get install. As well it fails and breaks install (G_RUN_CMD), if one package is not yet installed.
+ DietPi-Software | WireGuard: Do not force reboot after kernel updates, instead inform user via prompt, if no automated reboot is triggered.
+ DietPi-Software | WireGuard: Do not reinstall wireguard-dkms to rebuild the kernel module, instead use dpkg-reconfigure, and, only if it was no fresh install.
+ DietPi-Software | WireGuard: Estimate required reboot by checking modprobe exit code; Enable IP forwarding and start service for current session only, if WireGuard module has been successfully enabled
+ DietPi-Software | WireGuard: Do not overwrite any existing keys or configs
+ DietPi-Software | WireGuard: Enable IPv6 tunnel and forwarding
+ DietPi-Software | WireGuard: Add comments to default client config about how to tunnel local network or server access only, DNS server choice and multiple clients
+ DietPi-Software | WireGuard: Allow choice between server and client setup. In case of client setup, skip all config steps, instead inform user to follow VPN provider instructions. Add VPN (auto)start info as well, in case the VPN provider does not include it.
+ CHANGELOG | WireGuard VPN is now available for installation.

MichaIng committed:
v6.20

+ DietPi-Set_Hardware | "make" package required, before we can use "make" command the build i-sabre-k2m kernel module

MichaIng committed:
v6,20

+ DietPi-WiFi-Monitor | Update network info on service start
+ DietPi-WiFi-Monitor | Check for valid WiFi adapter/interface on service start, otherwise exit
+ DietPi-WiFi-Monitor | Check for valid WiFi adapter/interface, before attempting reconnection. In case assume WiFi adapter has been unplugged and exit
+ DietPi-WiFi-Monitor | Mute ping output: "-q" still shows init and result output, mutes single ping attempts only. Forward all to /dev/null to make it totally silent

MichaIng committed:
v6.20

+ DietPi-WiFi-Monitor | Add network info update and valid wlan[0-9] interface check to the systemd units ExecStartPre, to assure the service start fails. Type=simple/exec does not fail to start, if error occurs within ExecStart command/script and Type=forking delays further service startups until the script has finished => unlimited or failure

MichaIng committed:
v6.20

+ DietPi-Logclear | Remove internally rotated LetsEncrypt log files, prior to loop
+ DietPi-Logclear | Create $TEMP_FILE in scripts working dir
+ DietPi-Logclear | Minor coding

MichaIng committed:
v6.20

+ CHANGELOG | DietPi-RAMlog/Logclear: The internally rotated LetsEncrypt (CertBot) log files are now removed, hourly with DietPi-RAMlog enabled, or when executing DietPi-Logclear manually.

MichaIng committed:
v6.20

+ CHANGELOG | Resolved an issue, where DietPi-WiFi-Monitor stuck in a reconnection loop, if no valid WiFi interface was detected or the WiFi adapter is unplugged.

MichaIng committed:
v6.20

+ DietPi-Cloudshell | Check for CPU governor and frequency /sys/* file existence, before attempting to read them, use 'N/A' as meaningful value otherwise.
+ DietPi-Cloudshell | Minor coding

MichaIng committed:
v6.20

+ CHANGELOG | DietPi-Cloudshell: Resolved an issue where CPU scene failed on VM and other devices that do not support CPU governors

MichaIng committed:
v6.20

+ DietPi-NordVPN | Safe settings file in a shell source-able way
+ DietPi-NordVPN | Minor coding

MichaIng committed:
v6.20

+ DietPi-NordVPN | Tiny: Reduce settings file permissions to minimum: Readable is enough to source file, no execution permissions required

MichaIng committed:
v6.20

+ DietPi-Software | MPD: Create /usr/local/etc/mpd.conf symlink based on "command -v mpd" path instead of checking distro version: Failsafe!
+ DietPi-Software | MPD: Remove mpd.conf symbolic link before real file, otherwise [[ -f/-e ]] are false both, only [[ -L ]] would be true, but not include manually placed real file
+ DietPi-Software | Minor alignment

Commit:
v6.20

+ Minor

Commit:
Merge branch 'dev' of https://github.com/Fourdee/DietPi into dev

MichaIng committed:
v6.20

+ DietPi-NordVPN | Failsafe: Wrap settings into single quotes to prevent issues with magic characters and spaces + escape single quotes within user and password strings.

MichaIng committed:
v6.20

+ CHANGELOG | tiny

MichaIng committed:
v6.20

+ RC up

MichaIng commented on dietpi/dietpi-cleaner:
```diff
@@ -24,9 +24,12 @@
 	#Import DietPi-Globals ---------------------------------------------------------------
 
+	#Exit trap (dietpi-globals)
+	G_EXIT_CUSTOM(){ unset aEnabledCleaners; }
```
@Fourdee 
However, I am still not sure if this has any effect or all allocated RAM usage is anyway freed, when the script exits.

https://unix.stackexchange.com/a/360377
- The TO mixes things up there, but the answer states as well:
> In any case, the variables themselves will be disposed of when the script ends and the operating system reclaims the shell's memory.
> ...
> There is no need to blank or unset variables at the end of a shell script in any way.

And a second one: https://stackoverflow.com/a/12998828

Fourdee replied:
@MichaIng 

Interesting, good find 👍 
So we only need to delete vars during runtime (if required).

MichaIng replied:
@Fourdee 
Jep seems like. I mean also even large arrays take a tiny amount of RAM, compared to the whole shell, but especially in large scripts where variables might be used doubled (e.g. dietpi-software/config) I would consider it still good practice to unset any used variable, after used.

MichaIng committed:
v6.20

+ DietPi-Config | Remove wireless-power setting from /etc/network/interfaces, if not supported by adapter/firmware

MichaIng committed:
v6.20

+ CHANGELOG | Resolved a visual-only issue, where WiFi interface starts throw an error message about unsupported power management

MichaIng committed:
v6.20

+ DietPi-Config | Syntax

MichaIng committed:
v6.20

+ DietPi-Patch | Remove wireless-power setting from /etc/network/interfaces, if not supported by adapter/firmware

Commit:
v6.20

- DietPi-Software | AmiBerry: Updated to v2.24, thanks @midwan! Brings several bug fixes, new features and improvements over the previous version (2.19). Please check the version history for full details on the changes made: https://github.com/midwan/amiberry/wiki/Version-history

Commit:
Merge branch 'dev' of https://github.com/Fourdee/DietPi into dev

Fourdee committed:
Merge pull request #2431 from Fourdee/nordvpn

DietPi-NordVPN | Safe settings file in a shell source-able way, tiny fix and minor coding

Commit:
v6.20

+ Syntax
+ Fix settings file: https://github.com/Fourdee/DietPi/pull/2431#issuecomment-456384688

Commit:
v6.20

+ split save file: https://github.com/Fourdee/DietPi/pull/2431#issuecomment-456385579

MichaIng committed:
v6.20

+ DietPi-WiFi-Monitor | Start after DietPi-Boot, to not have two "obtain_network_details" calls interfere
+ DietPi-WiFi-Monitor | Type=simple not required, since it's default

MichaIng committed:
v6.20

+ DietPi-WiFi-Monitor | Requires DietPi-RAMdisk, since it sources DietPi-Globals and reads /DietPi/dietpi/.network

Commit:
v6.20

+ Always wait for dietpi-postboot before running autostarts
+ AmiBerry use service to start it

Commit:
Merge branch 'dev' of https://github.com/Fourdee/DietPi into dev

MichaIng committed:
v6.20

+ DietPi-Patch | Avoid doubled Amiberry reinstall when updating from pre-v6.8 to post-v6.19
+ DietPi-Patch | Update reinstall table info

MichaIng committed:
v6.20

+ DietPi-Software | Homeassistant: Remove tripled libssl-dev from dependencies, git (installed via aSOFTWARE_REQUIRES_GIT) and curl + sudo DietPi core packages
+ DietPi-Software | Homeassistant: As reference add install dependencies as comment to uninstall part, check for files/dirs/users/groups, before attempting removal, to avoid error messages

MichaIng committed:
v6.20

+ DietPi-Launcher | Add DietPi-NordVPN, to Misc section

MichaIng committed:
v6.20

+ DietPi-CPU_info | Allow VMs to run the script, throw info, that most info will be N/A. This allows as well to test/check syntax and output format on virtual testing machines.
+ DietPi-CPU_info | Estimate scaling frequencies in separate function. Speeds up both input modes
+ DietPi-CPU_info | Minor coding

MichaIng committed:
v6.20

+ DietPi-CPU_info | Optimize VM info
+ DietPi-CPU_info | Show CPU freq table headers only, if frequencies are available

MichaIng committed:
v6.20

+ DietPi-CPU_info | Temperature: If no INT found, do not expect °C, simply show G_OBTAIN_CPU_TEMP() 'N/A' output, instead of KERNELERR warning, if no interface files were found

Commit:
v6.20

+ No need to delete vars/array during script exit: https://github.com/Fourdee/DietPi/pull/2432#pullrequestreview-194556366

Commit:
Merge branch 'dev' of https://github.com/Fourdee/DietPi into dev

Commit:
v6.20

- DietPi-Software | LXDE: Installation will no longer enable the weather application. This is due to Yahoo API hard coded into LXpanel, being taken offline: https://github.com/Fourdee/DietPi/issues/2416#issuecomment-456805925

MichaIng committed:
META | dietpi-survey_report: Minor coding

+ No double quotes required (to preserve spaces and avoid splitting) when assigning variables to variables

Fourdee commented:
Ok, will push this today, just want to run some local testing 1st.

Commit:
v6.20

+ Syntax fixes + tweaks.

Commit:
Merge branch 'dev' of https://github.com/Fourdee/DietPi into dev

Commit:
v6.20

+ Tweak.

Commit:
v6.20

+ Tweaks:  https://github.com/Fourdee/DietPi/commit/9bf02d9b01db4af212c4b0dfcb3947e6b9407cab#commitcomment-32050548

Commit:
v6.20

+ Return 0 if not supported by the device.

Fourdee commented:
@MichaIng 

> Also it forcefully adds the comment, if not setting or comment is presen

👍 
Not ideal, but its fine, no harm in adding the entry as commented out.

I also like the visual output of ```G_CONFIG_INJECT```, shows whats being done for the end users that want to see it.

Commit:
v6.20

+ Minor
